### ISSUE TITLE
Batch Decisions

### ISSUE THREAD

**Issue by kylemath** [kylemath/eeg-notebooks#23]
So, batches... in an ideal world, you would do gradient descent on each data point. But this would take a long time, and by batching things up you can save time by doing things in parallel.
So, a batch is a collection of data points that you run through in parallel.
and this makes all the calculations pool over a set of data.
Often you use a small number for the batch size, as this gets you closer to the single data point technique.
But there is a lot of research now to use really really big batches.
the bigger the batch, the more variance you wash out.
And the less learning signal you get... so it is a bias (to the error in the batch) versus variance tradeoff
So, ideal would be batch_size = 1 but that means that the learning is slow... 
As you increase (and you want to increase in powers of 2 (.. 4, 8, 16, ...  for memory layout efficiency))
The ideal batch size would be as small as you can go training time wise
